py/server/deephaven/table.py

```diff
@@ -1656,3 +1664,849 @@ def partitioned_transform(self, other: PartitionedTable, func: Callable[[Table,
+
+    @auto_locking_op
+    def snapshot(self, source_table: Union[Table, PartitionedTableProxy], do_init: bool = False,
+                 cols: Union[str, List[str]] = None) -> PartitionedTableProxy:
+        """Applies the :meth:`~Table.snapshot` table operation to all constituent tables of the underlying
+        partitioned table with the provided source table or PartitionedTableProxy, and produces a new
+        PartitionedTableProxy with the result tables as the constituents of its underlying partitioned table.
+
+        In the case of source table being another PartitionedTableProxy, the :meth:`~Table.snapshot` table operation
+        is applied to the matching pairs of the constituent tables from both underlying partitioned tables.
+
+        Note, the constituent tables are often time tables that add new rows at a regular, user-defined interval.
+
+        Args:
+            source_table (Union[Table, PartitionedTableProxy]): the table or PartitionedTableProxy to be snapshot
+            do_init (bool): whether to snapshot when this method is initially called, default is False
+            cols (Union[str, List[str]]): names of the columns of the constituent table to be included in the snapshot,
+                default is None, meaning all the columns
+
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            cols = to_sequence(cols)
+            table_op = jpy.cast(source_table.j_object, _JTableOperations)
+            return PartitionedTableProxy(j_pt_proxy=self.j_pt_proxy.snapshot(table_op, do_init, *cols))
+        except Exception as e:
+            raise DHError(e, "snapshot operation on the PartitionedTableProxy failed.") from e
+
+    def sort(self, order_by: Union[str, Sequence[str]],
+             order: Union[SortDirection, Sequence[SortDirection]] = None) -> PartitionedTableProxy:
+        """Applies the :meth:`~Table.sort` table operation to all constituent tables of the underlying partitioned
+        table, and produces a new PartitionedTableProxy with the result tables as the constituents of its underlying
+        partitioned table.
+
+        Args:
+            order_by (Union[str, Sequence[str]]): the column(s) to be sorted on
+            order (Union[SortDirection, Sequence[SortDirection], optional): the corresponding sort directions for
+                each sort column, default is None, meaning ascending order for all the sort columns.
+
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            order_by = to_sequence(order_by)
+            if not order:
+                order = (SortDirection.ASCENDING,) * len(order_by)
+            order = to_sequence(order)
+            if len(order_by) != len(order):
+                raise DHError(message="The number of sort columns must be the same as the number of sort directions.")
```

Comment:
Making sure that our sort operations cover something like:
```
t.sort(order_by=["A","B",C"], order=SortDirection.DESCENDING)
```
where order is a single item that gets duplicated over the 3 columns.  ... or did we decide not to do this because someone could just do:
```
t.sort(order_by=["A","B",C"], order=[SortDirection.DESCENDING]*3)
```